apis/database/v1beta1/nosql_types.go

```diff
@@ -0,0 +1,192 @@
+/*
+Copyright 2020 The Crossplane Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1beta1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/Azure/azure-sdk-for-go/services/cosmos-db/mgmt/2015-04-08/documentdb"
+	runtimev1alpha1 "github.com/crossplane/crossplane-runtime/apis/core/v1alpha1"
+	"github.com/crossplane/crossplane-runtime/pkg/resource"
+	"github.com/pkg/errors"
+
+	apisv1alpha3 "github.com/crossplane/provider-azure/apis/v1alpha3"
+)
+
+// Error strings
+const (
+	errResourceIsNotValidNoSQL = "the managed resource is not CosmosDB"
+)
+
+// ResourceGroupNameReferencerForCosmosDBAccount is an attribute referencer that
+// resolves the name of a the ResourceGroup.
+type ResourceGroupNameReferencerForCosmosDBAccount struct {
+	apisv1alpha3.ResourceGroupNameReferencer `json:",inline"`
+}
```

Comment:
https://github.com/crossplane/provider-azure/pull/137/files#diff-824db1026f7a9936a86c5dc9029493f8R29

We've just this week started replacing this referencer pattern with a new (and hopefully less complicated) API. You can see an example of the new API in the above PR. You're welcome to leave this PR as is - the method you're using is still supported but will be marked deprecated in the next release of crossplane-runtime.

If you do want to use the new pattern you'll need to:

1. Rebase on master to update your crossplane-runtime dependency.
1. Change the type of `ResourceGroupNameRef` to `*runtimev1alpha1.Reference`
1. Add a corresponding `ResourceGroupNameSelector` of type `runtimev1alpha1.Selector`
1. Add a `ResolveReferences` method to `*CosmosDBAccount` (it can be identical to the one linked above)
1. Pass `managed.WithReferenceResolver(managed.NewAPISimpleReferenceResolver(mgr.GetClient()))` to `managed.NewReconciler` in your controller's `Setup` function.

Reply:
Err, actually, I think you will need to make this change because my PR deleted `apisv1alpha3.ResourceGroupNameReferencer` 😬

Reply:
NP. Done.